src/include/function/unary_function_executor.h

```diff
@@ -89,6 +89,28 @@ struct UnaryUDFFunctionWrapper {
     }
 };
 
+struct CastChildFunctionExecutor {
+    template<typename OPERAND_TYPE, typename RESULT_TYPE, typename FUNC, typename OP_WRAPPER>
+    static void executeSwitch(
+        common::ValueVector& operand, common::ValueVector& result, void* dataPtr) {
+        // this vector is of var list type and the child vector is of non-nested types then cast
+        KU_ASSERT(operand.dataType.getLogicalTypeID() == common::LogicalTypeID::VAR_LIST &&
```

Comment:
This executor looks incorrect to me:
1. The executor doesn't handle nulls correctly. We should loop through each value in the operand and set null in the result accordingly.
2. If the operand state is selected, we should skip casting unselected elements.